contracts/LiquidityBridgeContract.sol

```diff
@@ -301,35 +301,38 @@ contract LiquidityBridgeContract {
             delete callRegistry[quoteHash];
             emit BridgeCapExceeded(quoteHash, transferredAmountOrErrorCode);
             return transferredAmountOrErrorCode;
-        }        
+        }
+
+        // the amount is safely assumed positive because it's already been validated in lines 287/298 there's no (negative) error code being returned by the bridge.
+        uint transferredAmount = uint(transferredAmountOrErrorCode);        
 	
         if (callRegistry[quoteHash].timestamp > 0) {
             uint refundAmount;
 
             if (callRegistry[quoteHash].success) {
-                refundAmount = min(uint(transferredAmountOrErrorCode), quote.value + quote.callFee);
+                refundAmount = min(transferredAmount, quote.value + quote.callFee);
             } else {
-                refundAmount = min(uint(transferredAmountOrErrorCode), quote.callFee);
+                refundAmount = min(transferredAmount, quote.callFee);
             }
             increaseBalance(quote.liquidityProviderRskAddress, refundAmount);
-            int256 remainingAmount = transferredAmountOrErrorCode - int(refundAmount);
+            uint remainingAmount = transferredAmount - refundAmount;
             
-            if (remainingAmount > dust) {
+            if (remainingAmount > uint(dust)) {
```

Comment:
I think that this casting is unnecessary

Reply:
removed.